File: src/org/r3/db/cms/carddescr/CardDescrDTO.java
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//


package org.r3.db.cms.carddescr;

public class CardDescrDTO extends CardDescrDTOBase
{
    public CardDescrDTO()
    {
        super();
    }

    public CardDescrDTO(CardDescrDTO another)
    {
        super(another);
    }

    public boolean hasPendingChanges()
    {
    	if (this.isSummaryDirty() || this.isTextbodyDirty() || this.isTitleDirty())
    		return true;
    	
    	return false;
    }
}

File: WebContent/org/r3/db/cms/carddescr/CardDescrDTO.js
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//


jQuery.r3Utils.namespace("org.r3.db.cms.carddescr");

org.r3.db.cms.carddescr.CardDescrDTO = Class.extend(
{
	 init: function(aDataBean)
	 {  
	 	 	 this.dtrowclass   = "grade_carddescr";
       this.dtokey       = aDataBean.dtokey   || null;
       this.dtodescr     = aDataBean.dtodescr || null;
       
        this.site = aDataBean.site || null;
        this.cardid = aDataBean.cardid || null;
        this.languagecode = aDataBean.languagecode || null;
        this.title = aDataBean.title || null;
        this.summary = aDataBean.summary || null;
        this.textbody = aDataBean.textbody || null;
        this.recstatus = aDataBean.recstatus || null;
        this.rectimestamp = aDataBean.rectimestamp || null;
 
	 }
    
  ,compareTo: function(anItemId)
   {
   	   if (!this.dtokey)
   	       return -99;
   	       
   	   if (this.dtokey < anItemId)
   	       return -1;
   	   else if (this.dtokey > anItemId)
   	   	   return  1;
   	   	   
   	   return 0;
   }
   
});
